client/state.cpp

```diff
@@ -234,16 +231,15 @@ bool State::update_frame(const torchcraft::fbs::FrameData* fd) {
   if (fd->data() == nullptr)
     return false;
   if (fd->is_diff()) {
-    this->frame_string = "";
     std::istringstream ss(std::string(
         reinterpret_cast<const char*>(fd->data()->data()), fd->data()->size()));
     replayer::FrameDiff diff;
     ss >> diff;
     replayer::frame_undiff(this->frame, this->frame, &diff);
   } else {
-    this->frame_string = std::string(
+    auto frame_string = std::string(
         reinterpret_cast<const char*>(fd->data()->data()), fd->data()->size());
-    std::istringstream ss(this->frame_string);
+    std::istringstream ss(frame_string);
```

Comment:
This will copy the frame string two times right now. This should bring you down to one (not sure if the compiler is smart enough to do it itself):

```c++
std::istringstream ss(std::string(
          reinterpret_cast<const char*>(fd->data()->data()), fd->data()->size());		          reinterpret_cast<const char*>(fd->data()->data()), fd->data()->size());
```

And this should bring you down to zero but requires some extra code (bad naming maybe):

```c++
class Membuf : public std::streambuf {
 public:
  Membuf(const char* data, size_t size) {
    setg(data, data, data + size);
  }
};

Membuf mbuf(reinterpret_cast<const char*>(fd->data()->data()), fd->data()->size());
std::istream ss(&mbuf);
```

Reply:
Thanks Jonas. This will be ripped out in an upcoming PR when we move to FlatBuffers and ditch streams entirely: https://github.com/TorchCraft/TorchCraft/blob/binary-frame-serialization-2/client/state.cpp#L233

Would you have any opposition to leaving this as-is for the moment?

Reply:
Ok then, no worries!